code-challenges.test.js

```diff
@@ -14,39 +14,108 @@
+    { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+    { name: "arthur dent", occupation: "a radio employee" }
+  ]
+  it("Returns an array with names capitalized", () => {
+    // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+    expect(CapName(hitchhikersCharacters)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
+//GOODFAILURE ReferenceError: CapName is not defined
 
 // b) Create the function that makes the test pass.
-
 // Pseudo code:
+//Function Name: CapName
+//Input: Array of objects
+//Output: Array of strings
+//Process: Create your function that takes in a parameter of an array. Use .map to iterate through the array and also return a new array. To call on a value from an object use the key:value. Since we know the first name is at index zero you should capitalize whatever character is at index 0. Split the value into an array, make sure that all the characters are seperated. Use .indexOf to find where the space is between the first and last name, this will help capitalize the last name. Then seperate the first and last name, have a variable hold each name. To do that make an array that seperates the first and last name by using .split(' '). That will make an array have two values. Then use concat (+) to combine the capital letter with the word. Make sure to use slice to get rid of the lowercase we don't need. Then make a variable that will hold the names together with concat (+). Lastly, return the string interpolation!
+
+const CapName = (arr) => {
+  return arr.map((value) => {
+    const firstCase = value.name.charAt(0).toUpperCase() //Capitalizes the first letter of the first name
+    const name = value.name.split('').indexOf(' ') // Finds the indexOf where the space is in each name. 
+    const lastCase = value.name.charAt(name + 1).toUpperCase() //Capitalizes the first letter of the last name
+
+    const seperateName = value.name.split(' ') //Makes the first name be in the 0 index while the last name is in the 1st index. 
+    const firstName = seperateName[0] // Holds just the 0 index of the array, so just the first name.
+    const lastName = seperateName[1] //Holds just the 1st index of the array, so just the last name.
+
+    const capFirstName = firstCase + firstName.slice(1) //Concats the capital letter of the first name and combines it with the first name. The slice uses everything after the 1st index.
+    const capLastName = lastCase + lastName.slice(1) //Concats the capital letter from the last name and combines it with the last name. The slice uses everything after the 1st index.
+    const fullName = capFirstName + ' ' + capLastName //Connects both the full capitalized first and last name
+
+    return `${fullName} is ${value.occupation}.` //Returns the string interpolation
```

Comment:
I saw your comment about refactor...yes 🤣
In all seriousness, if this is how you need to put it to understand it, that's fine. You will continue to learn and grow, and over time the refactors will come. Soon you'll be able to look back at this and refactor it no problem. I'll give you another approach you can take a look at, like I gave Tori:

![Screenshot 2023-10-15 at 1 16 27 PM](https://github.com/learn-academy-2023-foxtrot/week-6-assessment-Varelazi/assets/106039699/bd216f61-91b0-4f80-a761-785cd5bcd53a)